pulpcore.te

```diff
@@ -177,3 +178,9 @@ optional_policy(`
 
 	pulpcore_filetrans_named_content(init_t)
 ')
+
+require {
+	type kernel_t;
+	type pulpcore_server_t;
+	class unix_dgram_socket { connect create sendto };
+}
```

Comment:
Is it really correct to have these requires here at the bottom? I think it's convention to place them at the top.

Reply:
I am still learning the policy syntax, and had reservations against that:

1. It seems like kernel_t is part of an external policy, so it should be in a require block. Correct?
https://selinuxproject.org/page/PolicyStatements#require

2. Will the syntax work if I put these 3 lines in the main block up top with the declarations? The top of the file is the declarations. But this block of  3 lines seem different than a declaration. It seems like the type lines set the subject, and the "class" line says what actions those subjects can do to what class of objects. It also seems different from the "pulpcore/pulpcore_server local policy" lines that specify the subject in the same line as the object and actions.

Reply:
Wait, are you talking about moving the entire require block up? Or putting the lines within the require block in the main body of the file, at the top?

Reply:
I was talking about moving the entire `require` block to the top.